src/Nancy/Responses/EmbeddedFileResponse.cs

```diff
@@ -47,19 +48,17 @@ private Stream GetResourceContent(Assembly assembly, string resourcePath, string
         {
             var resourceName = assembly
                 .GetManifestResourceNames()
-                .Where(x => GetFileNameFromResourceName(resourcePath, x).Equals(name, StringComparison.OrdinalIgnoreCase))
-                .Select(x => GetFileNameFromResourceName(resourcePath, x))
-                .FirstOrDefault();
+                .FirstOrDefault(x => GetFileNameFromResourceName(resourcePath, x).Equals(name, StringComparison.OrdinalIgnoreCase));
 
-            resourceName =
-                string.Concat(resourcePath, ".", resourceName);
+            if (resourceName == null)
+                return null;
 
             return assembly.GetManifestResourceStream(resourceName);
         }
 
         private static string GetFileNameFromResourceName(string resourcePath, string resourceName)
         {
-            return resourceName.Replace(resourcePath, string.Empty).Substring(1);
+            return Regex.Replace(resourceName, resourcePath, string.Empty, RegexOptions.IgnoreCase).Substring(1);

```

Comment:
@damianh I highly doubt that. Serving static content from embedded resources is not fast by itself, so a few additional milliseconds will not make a difference here. Otherwise, we'll have to invent our very own

    public static string Replace(this string source, string oldValue, string newValue, 
        StringComparison comparisonType)